core/tests/test_views.py

```diff
@@ -177,3 +211,29 @@ def test_add_deeply_nested_comment_valid(client: Client, another_user: User, pos
     assert post.children_count == 10
     assert parent_comment.children_count == 0
     assert parent_comment.parent.children_count == 1
+
+
+@pytest.mark.django_db()
+def test_post_user_avatar_display(client: Client, community: Community, user_with_avatar: User) -> None:
+    data = {
+        "community": community.pk,
+        "title": "Test Post Title",
+        "content": "This is a test post content.",
+    }
+    client.force_login(user_with_avatar)
+    response = client.post(reverse("post-create"), data=data, follow=True)
+    post = response.context["post"]
+    assert post.author.avatar.url == user_with_avatar.avatar_url
+
+
+@pytest.mark.django_db()
+def test_post_user_without_avatar(client: Client, community: Community, user: User, default_avatar_url: str) -> None:
+    data = {
+        "community": community.pk,
+        "title": "Test Post Title",
+        "content": "This is a test post content.",
+    }
+    client.force_login(user)
```

Comment:
I would still simply create this post in DB by post.objects.create but ok. 
If you are going this way, please show the post was created or there are no errors on page. One of two ways
- count posts objects
- check context form errrors.